ProjBobcat/ProjBobcat/Class/Helper/SystemInfoHelper.cs

```diff
@@ -38,64 +43,30 @@ public static MemoryInfo GetMemoryUsage()
     public static async IAsyncEnumerable<string> FindJava(bool fullSearch = false)
     {
         var result = new HashSet<string>();
+        var DiskName = getDisk();
 
         if (fullSearch)
             await foreach (var path in DeepJavaSearcher.DeepSearch())
                 result.Add(path);
-        //C遍历Java常见位置
-        try
-        {
-            DirectoryInfo TheFolder = new DirectoryInfo("C:\\Program Files\\Java");
-            foreach (DirectoryInfo NextFolder in TheFolder.GetDirectories())
-            {
-                string FullPath = "C:\\Program Files\\Java\\" + NextFolder.Name + "\\bin\\javaw.exe";
-                if (File.Exists(FullPath))
-                    result.Add(FullPath);
-            }
-        }
-        catch { }
-
-        try
-        {
-            DirectoryInfo TheFolder = new DirectoryInfo("D:\\Program Files\\Java");
-            foreach (DirectoryInfo NextFolder in TheFolder.GetDirectories())
-            {
-                string FullPath = "D:\\Program Files\\Java\\" + NextFolder.Name + "\\bin\\javaw.exe";
-                if (File.Exists(FullPath))
-                    result.Add(FullPath);
-            }
-        }
-        catch { }
-
-        try
+        
+        foreach(var JP in DiskName)
         {
-            DirectoryInfo TheFolder = new DirectoryInfo("E:\\Program Files\\Java");
-            foreach (DirectoryInfo NextFolder in TheFolder.GetDirectories())
+            try
             {
-                string FullPath = "E:\\Program Files\\Java\\" + NextFolder.Name + "\\bin\\javaw.exe";
-                if (File.Exists(FullPath))
-                    result.Add(FullPath);
+                DirectoryInfo TheFolder = new DirectoryInfo($"{JP}\\Program Files\\Java");
+                foreach (DirectoryInfo NextFolder in TheFolder.GetDirectories())
+                {
+                    string FullPath = JP+"\\Program Files\\Java\\" + NextFolder.Name + "\\bin\\javaw.exe";
+                    if (File.Exists(FullPath))
+                        result.Add(FullPath);
+                }
             }
+            catch { }
```

Comment:
Unnecessary trying & catching, you had better replace it with logical judgements which are more explicit.